File: rrs-server/src/main/java/RoomReservationSystem/service/ClassroomService.java
package RoomReservationSystem.service;

import RoomReservationSystem.dto.ClassroomDTO;
import RoomReservationSystem.exception.BuildingNotExistsException;
import RoomReservationSystem.exception.ClassroomAlredyExistsException;
import RoomReservationSystem.exception.ClassroomNotExistsException;
import RoomReservationSystem.model.Building;
import RoomReservationSystem.model.Classroom;

import java.util.List;

/**
 * Az osztálytermekkel kapcsolatos műveletekért felelős interfész Részletes
 * információ a függvényekről a megválósításnál
 *
 * @author Tomecz Patrik
 */
public interface ClassroomService {

    Classroom save(Classroom classroom) throws ClassroomAlredyExistsException;

    Classroom findByNameAndBuildingName(String name, String buildingName) throws ClassroomNotExistsException, BuildingNotExistsException;

    Classroom findById(int id) throws ClassroomNotExistsException;

    List<Classroom> findByName(String name);

    List<Classroom> findAll();

    List<Classroom> findByHasPc(boolean hasPC);

    List<Classroom> findByHasProjector(boolean hasProjector);

    List<Classroom> findByChairsLessThan(int num);

    List<Classroom> findByChairsGreaterThan(int num);

    List<Classroom> findByChairsBetween(int from, int to);

    List<Classroom> findByBuilding(Building building);

    List<Classroom> findByBuildingName(String buildingName) throws BuildingNotExistsException;

    List<String> getNamesByBuilding(String building) throws BuildingNotExistsException;

    Classroom findByDTO(ClassroomDTO classroomDTO) throws BuildingNotExistsException;

    boolean existsById(int id);

    boolean existsByNameAndBuilding(String name, Building building);

    void deleteByNameAndBuildingName(String name, String buildingName) throws ClassroomNotExistsException, BuildingNotExistsException;
}

File: rrs-client/src/app/services/classroom.service.ts
import { Injectable } from "@angular/core";
import { Routes } from "../config/routes.config";
import { Observable } from "rxjs/Observable";
import { HttpClient } from "@angular/common/http";
import { Classroom } from "../models/Classroom";

/**
 * Az tantermekhez tartozó service osztály
 */
@Injectable()
export class ClassroomService {
  constructor(private http: HttpClient) {}

  getAll(): Observable<Classroom[]> {
    return <Observable<Classroom[]>>(
      this.http.get(Routes.getUrl(Routes.CLASSROOM_GET_ALL))
    );
  }

  getNamesByBuilding(building: string): Observable<string[]> {
    return <Observable<string[]>>(
      this.http.get(
        Routes.getUrl(Routes.CLASSROOM_GET_NAMES_BY_BUILDING) +
          `?building=${building}`
      )
    );
  }

  findByName(roomName: string): Observable<Classroom> {
    return <Observable<Classroom>>(
      this.http.get(
        Routes.getUrl(Routes.CLASSROOM_FIND_BY_NAME) + "/" + roomName
      )
    );
  }

  findByBuildingName(buildingName: string): Observable<Classroom[]> {
    return <Observable<Classroom[]>>(
      this.http.get(
        Routes.getUrl(Routes.CLASSROOM_FIND_BY_BUILDING_NAME) +
          "/" +
          buildingName
      )
    );
  }

  findByNameAndBuildingName(
    name: string,
    buildingName: string
  ): Observable<Classroom> {
    return <Observable<Classroom>>(
      this.http.get(
        Routes.getUrl(Routes.CLASSROOM_FIND_BY_NAME_AND_BUILDING_NAME) +
          `?name=${name}&buildingName=${buildingName}`
      )
    );
  }

  findByHasPC(hasPC: boolean): Observable<Classroom[]> {
    return <Observable<Classroom[]>>(
      this.http.get(
        Routes.getUrl(Routes.CLASSROOM_FIND_BY_HAS_PC) + "/" + hasPC
      )
    );
  }

  findByHasProjector(hasProjector: boolean): Observable<Classroom[]> {
    return <Observable<Classroom[]>>(
      this.http.get(
        Routes.getUrl(Routes.CLASSROOM_FIND_BY_HAS_PROJECTOR) +
          "/" +
          hasProjector
      )
    );
  }

  findByChairsLessThan(chairs: number): Observable<Classroom[]> {
    return <Observable<Classroom[]>>(
      this.http.get(
        Routes.getUrl(Routes.CLASSROOM_FIND_BY_CHAIRS_LESS_THAN) + "/" + chairs
      )
    );
  }

  findByChairsGreaterThan(chairs: number): Observable<Classroom[]> {
    return <Observable<Classroom[]>>(
      this.http.get(
        Routes.getUrl(Routes.CLASSROOM_FIND_BY_CHAIRS_GREATER_THAN) +
          "/" +
          chairs
      )
    );
  }

  findByChairsBetween(from: number, to: number): Observable<Classroom[]> {
    return <Observable<Classroom[]>>(
      this.http.get(
        Routes.getUrl(Routes.CLASSROOM_FIND_BY_CHAIRS_BETWEEN) +
          `?from=${from}&to=${to}`
      )
    );
  }

  createClassroom(classroom: Classroom): Observable<Classroom> {
    return <Observable<Classroom>>(
      this.http.post(
        Routes.getUrl(Routes.CLASSROOM_CREATE_CLASSROOM),
        classroom
      )
    );
  }

  deleteByNameAndBuildingName(
    name: string,
    buildingName: string
  ): Observable<any> {
    return <Observable<any>>(
      this.http.delete(
        Routes.getUrl(Routes.CLASSROOM_DELETE_BY_NAME_AND_BUILDING_NAME) +
          `?name=${name}&buildingName=${buildingName}`
      )
    );
  }

  existsById(id: number): Observable<boolean> {
    return <Observable<boolean>>(
      this.http.get(Routes.getUrl(Routes.CLASSROOM_EXISTS_BY_ID) + `?id=${id}`)
    );
  }

  existsByNameAndBuilding(name: string, building: string): Observable<boolean> {
    return <Observable<boolean>>(
      this.http.get(
        Routes.getUrl(Routes.CLASSROOM_EXISTS_BY_NAME_AND_BUILDING) +
          `?name=${name}&building=${building}`
      )
    );
  }
}
